hail/python/test/hail/methods/test_qc.py

```diff
@@ -285,3 +289,105 @@ def test_charr(self):
+    def test_vep_grch38_consequence_true(self):
+        gnomad_vep_result = hl.import_vcf(resource('sample.gnomad.genomes.r3.0.sites.chr1.vcf.gz'), reference_genome='GRCh38', force=True)
+        hail_vep_result = hl.vep(gnomad_vep_result, csq=True)
+
+        assert gnomad_vep_result.select_rows(vep=gnomad_vep_result.info.vep.map(lambda x: x.split('|')[:8])).rows()._same(
+            hail_vep_result.select_rows(vep=hail_vep_result.vep.map(lambda x: x.split('|')[:8])).rows().drop('vep_csq_header'))
+
+        vep_csq_header = hl.eval(hail_vep_result.vep_csq_header)
+        assert 'Consequence annotations from Ensembl VEP' in vep_csq_header, vep_csq_header
+
+    @skip_unless_service_backend(clouds=['gcp'])
+    @set_gcs_requester_pays_configuration(GCS_REQUESTER_PAYS_PROJECT)
+    def test_vep_grch37_consequence_false(self):
+        mt = hl.import_vcf(resource('sample.gnomad.exomes.r2.1.1.sites.chr1.vcf.gz'), reference_genome='GRCh37', force=True)
+        hail_vep_result = hl.vep(mt, csq=False)
+        ht = hail_vep_result.rows()
+        ht = ht.select(variant_class=ht.vep.variant_class)
+        result = ht.head(1).collect()[0]
+        assert result.variant_class == 'SNV', result
+
+    @skip_unless_service_backend(clouds=['gcp'])
+    @set_gcs_requester_pays_configuration(GCS_REQUESTER_PAYS_PROJECT)
+    def test_vep_grch38_consequence_false(self):
+        mt = hl.import_vcf(resource('sample.gnomad.genomes.r3.0.sites.chr1.vcf.gz'), reference_genome='GRCh38', force=True)
+        hail_vep_result = hl.vep(mt, csq=False)
+        ht = hail_vep_result.rows()
+        ht = ht.select(variant_class=ht.vep.variant_class)
+        result = ht.head(1).collect()[0]
+        assert result.variant_class == 'SNV', result
+
+    @skip_unless_service_backend(clouds=['gcp'])
+    @set_gcs_requester_pays_configuration(GCS_REQUESTER_PAYS_PROJECT)
+    def test_vep_grch37_against_dataproc(self):
+        mt = hl.import_vcf(resource('sample.vcf.gz'), reference_genome='GRCh37', force=True)
+        mt = mt.head(20)
+        hail_vep_result = hl.vep(mt)
+        initial_vep_dtype = hail_vep_result.vep.dtype
+        hail_vep_result = hail_vep_result.annotate_rows(vep=hail_vep_result.vep.annotate(
+            input=hl.str('\t').join([hail_vep_result.locus.contig, hl.str(hail_vep_result.locus.position), ".", hail_vep_result.alleles[0], hail_vep_result.alleles[1], ".", ".", "GT"])
+        ))
+        hail_vep_result = hail_vep_result.rows().select('vep')
+
+        def reorder_lof_info(ht):
+            return ht.annotate(vep=ht.vep.annotate(
+                transcript_consequences=hl.map(lambda csq: csq.annotate(
+                    lof_info=hl.if_else(csq.lof_info == 'null', hl.null(hl.tdict(hl.tstr, hl.tstr)), hl.dict(
+                        hl.map(lambda kv: hl.tuple([kv.split(":")[0], kv.split(":")[1]]), csq.lof_info.split(','))))),
+                                               ht.vep.transcript_consequences)))
```

Comment:
Let's also get rid of this `hl.map` its too hard to see which argument matches function.

```python3
transcript_consequences=ht.vep.transcript_consequences.map(
    lambda csq: csq.annotate(
        lof_info=...
    )
)
```